lib/ansible/modules/net_tools/netplan.py

```diff
@@ -0,0 +1,903 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: netplan
+author:
+    - Luis Eduardo (@lets00)
+short_description: Manage network configurations using netplan.
+version_added: "2.8"
+description:
+    - This module allows to you manage network configurations using netplan.
+      It creates/edits a netplan YAML file and saves on /etc/netplan/ directory
+      and apply on system.
+
+requirements: [ PyYAML, netplan ]
+notes:
+    - Online netplan page U(https://github.com/CanonicalLtd/netplan/blob/master/doc/netplan.md)
+options:
+  filename:
+    description:
+      - Name of file that will be create or open to add or remove network
+        interfaces.
+    required: true
+
+  renderer:
+    description:
+      - Use the given networking backend for this definition. Currently
+        supported are networkd and NetworkManager. Default is networkd.
+    choices: [ networkd, NetworkManager ]
+    required: false
+
+  version:
+    description:
+      - The YAML currently being used
+    required: false
+    type: int
+
+  interface-id:
+    description:
+      - Interface ID that must be unique.
+    required: true
+
+  type:
+    description:
+      - Interface type. Depends on type, you can use specific params
+        (E.g bridges can use stp param).
+    choices: [ bridges, bonds, ethernets, vlans, wifis ]
+    required: true
+
+  state:
+    description:
+      - C(present) define a net interface on C(filename);
+        C(absent) undefine a net interface on C(filename).
+    choices: [ present, absent ]
+    required: true
+
+  dhcp4:
+    description:
+      - Enable/Disable DHCP for IPv4. Default is disabled.
+    required: false
+    type: bool
+
+  dhcp6:
+    description:
+      - Enable/Disable DHCP for IPv6. Default is disabled.
+    required: false
+    type: bool
+
+  link-local:
+    description:
+      - Configure the link-local addresses to bring up. Valid options are
+        ipv4 and 'ipv6', which respectively allow enabling IPv4 and IPv6
+        link local addressing.
+    choices: [ ipv4, ipv6 ]
+    required: false
+
+  critical:
+    description:
+      - Designate the connection as "critical to the system", meaning that
+        special care will be taken by systemd-networkd to not release the IP
+        from DHCP when it the daemon is restarted. Networkd backend only.
+    required: false
+    type: bool
+
+  dhcp-identifier:
+    description:
+      - When set to C(mac); pass that setting over to systemd-networkd to use
+        the device's MAC address as a unique identifier rather than a
+        RFC4361-compliant Client ID. This has no effect when NetworkManager
+        is used as a renderer.
+    required: false
+
+  accept-ra:
+    description:
```

Comment:
Clarify the documentation; something like this would be more concise and clearer:
```
Control IPv6 autoconfiguration using Router Advertisement.
```